src/frontends/pytorch/src/utils.cpp

```diff
@@ -142,6 +142,22 @@ element::Type convert_dtype(int64_t pt_type) {
     return TORCH_TO_OV_TYPE.at(pt_type);
 };
 
+Output<Node> apply_dtype(const NodeContext& context, int dtype_port, const Output<Node>& input_tensor) {
+    if (std::dynamic_pointer_cast<opset10::Constant>(
+            context.get_input_from_visible_context(dtype_port).get_node_shared_ptr())) {
+        auto dtype = convert_dtype(context.const_input<int64_t>(dtype_port));
+        return context.mark_node(std::make_shared<opset10::Convert>(input_tensor, dtype));
+
+    } else if (const auto& fw_node = cast_fw_node(context.get_input(dtype_port).get_node_shared_ptr(), "prim::dtype")) {
+        auto out_tensor = fw_node->input_value(0);
+        return context.mark_node(std::make_shared<opset10::ConvertLike>(input_tensor, out_tensor));
+
+    } else {
+        FRONT_END_OP_CONVERSION_CHECK(false, "Couldn't get dtype input");
+    }
```

Comment:
why can't we check if `context.get_input(dtype_port).get_element_type().is_static()` and use `Convert` if it is static? And use `ConvertLike` in all other cases?

Can you please see (may be it will be helpful) `ov::Output<ov::Node> create_same_type_const_scalar(const ov::Output<ov::Node>& same_type_output, T value)` approach from TF FE? Can we apply the same?

Reply:
it works slightly different, dtype can be represented as operation torch.dtype(tensor), which returns dtype of input tensor or can be represented by native torch dtype e.g. torch.float32, which turns to enum constant int and should be mapped to openvino dtype. As we do not have operations which extract dtype from node, we are trying to resolve torch.dtype with replacement convertlike with its input.

However, maybe I do not fully understand your suggestion, maybe it required offline discussion